lib/usb-device.js

```diff
@@ -0,0 +1,237 @@
+      for (let iface of interfaces) {
+        currentInterface = iface;
+        // adb interface will only have 2 endpoints
+        if (currentInterface.endpoints.length !== 2) continue;
+        // interface for ADB always has these values in it's descriptor
+        let interfaceDescriptor = currentInterface.descriptor;
+        if (interfaceDescriptor.bInterfaceClass !== ADB_VALUES.ADB_CLASS &&
+            interfaceDescriptor.bInterfaceSubClass !== ADB_VALUES.ADB_SUBCLASS &&
+            interfaceDescriptor.bInterfaceProtocol !== ADB_VALUES.ADB_PROTOCOL) {
+          continue;
+        }
+        currentInterface = iface;
+        //if we get to this point we should have the interface we want
+        break;
+      }
+      return currentInterface;
+    }
+  }
+
+  // loops through all devices found by libusb
+  // returns a libusb device that has an ADB interface or null
+  static findAdbDevice () {
+    let usbDevices = usb.getDeviceList();
+    for (let device of usbDevices) {
+      let deviceInterface = this.getAdbInterface(device);
+      if (deviceInterface !== null) {
+        console.log("Found an ADB device.");
+        return new USBDevice(device, deviceInterface);
+      }
+    }
+    return null;
+  }
+
+  // opens the device and sets up the endpoints for out object
+  async claimDevice () {
+    console.log("Trying to claim the device.");
+    // we need to call these in OSX, see nonolith/nodeusb issue#61
+    if (process.platform === 'darwin') {
+      this.device.__open();
+      this.device.__claimInterface(0);
+    }
+    this.device.open();
+    console.log("Device opened.");
+    this.deviceInterface = Promise.promisifyAll(this.deviceInterface);
+    this.deviceInterface.claim();
+    await this.deviceInterface.setAltSettingAsync(0);
+    let endpoints = this.deviceInterface.endpoints;
+    // set the endpoints for our USBDevice object
+    // note that these endpoints are on the device, so it's endpoint in
+    // is our endpoint out, the endpoint we want to send data to
+    if (endpoints[0] === usb.LIBUSB_ENDPOINT_IN) {
+      this.inputEndpoint = endpoints[1];
+      this.outputEndpoint = endpoints[0];
+    } else {
+      this.inputEndpoint = endpoints[0];
+      this.outputEndpoint = endpoints[1];
+    }
+    this.inputEndpoint.transferType = usb.LIBUSB_TRANSFER_TYPE_BULK;
+    this.outputEndpoint.transferType = usb.LIBUSB_TRANSFER_TYPE_BULK;
+    this.inputEndpoint.timeout = 2500;
+    this.outputEndpoint = Promise.promisifyAll(this.outputEndpoint);
+    this.inputEndpoint = Promise.promisifyAll(this.inputEndpoint);
+    return;
+  }
+
+  // handles authentication between our machine and
+  // the android device upon initial connection
+  // TODO: attempt to sign token sent to us
+  async handleAuth () {
+    let publicKeyBuf = new Buffer(publicKeyString.length + 1);
+    publicKeyString.copy(publicKeyBuf);
+    publicKeyBuf[-1] = 0;
+    let authMsg = generateMessage(ADB_COMMANDS.CMD_AUTH
```

Comment:
style note: we'd write this as:

```js
let authMsg = generateMessage(ADB_COMMANDS.CMD_AUTH,
                              3, 0, publicKeyBuf);
```

i.e., when you need to break a line, line up the next argument just after the paren on the first line.